zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -754,6 +766,20 @@ public void persist(AuthenticationInfo subject) throws IOException {
     repo.save(this, subject);
   }
 
+  public void rename(AuthenticationInfo subject) throws IOException {
+    Preconditions.checkNotNull(subject, "AuthenticationInfo should not be null");
+    stopDelayedPersistTimer();
+    snapshotAngularObjectRegistry(subject.getUser());
+    index.updateIndexDoc(this);
+    
+    FileInfo currentFile = this.getFileInfo().copy();
+    FileInfo newFile = FileInfo.createInstance();
+    // assuming in the same folder
+    newFile.setFolder(currentFile.getFolder());
+    newFile.setFile(Util.convertTitleToFilename(this.getName()));
+    this.setFileInfo(repo.rename(currentFile, newFile, subject));
```

Comment:
Is the FileInfo used to store the original note title ? If it is, then I think just one filed `oldName` should be sufficient. The class `FileInfo` is not a very clear concept for me what it is used for.